ocaml/typing/mode.ml

```diff
@@ -1362,9 +1360,8 @@ module Monadic = struct
     match submode m0 m1 with
     | Ok () -> Ok ()
     | Error { left = uni0, (); right = uni1, () } ->
-      if Uniqueness.Const.le uni0 uni1
-      then assert false
-      else Error (`Uniqueness { left = uni0; right = uni1 })
+      assert (not (Uniqueness.Const.le uni0 uni1));
+      Error (`Uniqueness { left = uni0; right = uni1 })
```

Comment:
The style in which it was originally written has a particular reason: as we add more axes (contention), it will be a chain of if-then-else. The original code is the special case where the chain length = 1. This minimizes the diff of the future PR introducing contention.

Reply:
Yes, makes sense. Please revert (I'm off to a meeting in a moment), and then merge when CI is green. Thanks.